client/asset/interface.go

```diff
@@ -71,6 +74,15 @@ type WalletConfig struct {
 	// DataDir is a filesystem directory the the wallet may use for persistent
 	// storage.
 	DataDir string
+	// EncXPrivKey is the extended private key for account 0 of the asset the
+	// wallet is providing. (m/44'/{assetID}'/0') It is encrypted using the wallet
+	// password. This config only needs to be used for internal wallets.
+	EncXPrivKey []byte
+	// XPubKey is the extended public key for account 0 of the asset the
+	// wallet is providing. (m/44'/{assetID}'/0') This is needed so that a locked
+	// wallet can give information about addresses/accounts. This config only
+	// needs to be used for internal wallets.
+	XPubKey string
```

Comment:
What we're actually interested in is determinism. Some wallets such as btc are deterministic based on supplying the same random seed as an input to wallet creation, where it appears that for geth, you just import the private key. So what we actually need to pass in is a deterministic seed. I agree with the way you derive the private key from the seed, but it should be done inside of **eth**. 

Just fyi, I will be introducing two new methods on `asset.Driver`, `Create` and `Exists`, that will be used to initialize "seeded" wallets, e.g. those with `IsInternalWallet = true`. So you won't pass the seed to `asset.Setup` as part of the `WalletConfig`. You'll just pass it once to the `(*Driver).Create` method.

Reply:
I was considering the extended private key that is derived from the app seed using the path `m/44'/{assetID}'/0'` to be the seed for each of the wallets. From this wallet seed, I derive the actual private key used for signing transactions inside of eth. Is the naming I have confusing, or do you think we should pass the app seed into the wallet backends?

Reply:
The is the right derivation for the key, but when I initialize a btc or dcr wallet, I give the seed and the derivation is done internaly to **btcwallet** or **dcrwallet**. See, for example in dcrwallet, `wallet.Create` → `udb.Initialize` → `createAddressManager` → `deriveCoinTypeKey`. And since we have built-in Bitcoin (and likely Decred) wallets coming any day now, I know we need the seed itself there. 

**eth** will need to handle the key derivation internally.

Reply:
Thanks for clarifying all that @buck54321.  Only thing I would add is that some time in the future I can two different asset-dependent seeding approaches - one for backends like dcrwallet and btcwallet that are good with a seed that is not an extended key at all, and one for assets where it's more convenient to just get an extended key or maybe a vanilla private key.  For now a unified approach is best where each asset takes the seed (not a key) and uses it's own approach to generating keys from that.

On the topic of eth accounts, one thing I'd like to see it the ability to get an account private key out of the eth package that is suitable for import into common Ethereum wallets such as MetaMask.  The internal {0,0} derivation path is probably irrelevant here unless we do intend to generate multiple accounts.  I mention this functionality because I would also like the ability to get the dcr and btc seeds so I can restore those wallets externally, but with eth the seed is meaningless without replicating whatever the eth package does internally to get the private key from a seed.

Reply:
Actually, @jrick recommends BLAKE256 (or BLAKE2 or SHA3, but we already import `github.com/decred/dcrd/crypto/blake256`).

Reply:
BLAKE256(appseed|assetid) makes sense in the context that not all wallets will be able to run with an account key. It's also simple and easy to reproduce.

Reply:
A agree also that BLAKE256(appseed|assetid) makes sense.

It looks like there's a good bit of overlap with your PR @buck54321 , so I'm thinking of maybe putting those new `asset.Driver` methods into this PR. This PR would assume that there is only a single wallet type for each asset, and it may be either seeded or not. Seeded wallets have the extra step of being created by calling `asset.Create`. Then your PR would add the ability to have multiple types of wallets for the same asset. Do you think this makes sense, or should I hold off on finishing this until you complete your PR?

Regarding giving the ability to export a seed and importing into an external wallet, the metamask default derivation path is `m/44'/0'/0`, but metamask also supports the ledger live default path which is `m/44'/0'/0/0`. I think either of these would be fine.

Reply:
> Seeded wallets have the extra step of being created by calling asset.Create. Then your PR would add the ability to have multiple types of wallets for the same asset. Do you think this makes sense, or should I hold off on finishing this until you complete your PR?

That sounds about right.

I'll be opening Bitcoin SPV by Monday(ish), and the `Driver` bits are implemented there. You're welcome to do it here so we can get it in earlier though.